app/actions/requests/CcaWrappedRequest.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package actions.requests
+
+import models.Accounts
+
+trait CcaWrappedRequest {
+  def isLoggedIn: Boolean
+  def optAccounts: Option[Accounts]
+  def yourDetailsName: Option[String]
+
+  def getYourDetailsName(optAccounts: Option[Accounts]): Option[String] = optAccounts.map { acc =>
```

Comment:
would this not be better to evaluate the value here, instead of having a method you call to enrich the wrapped request.